src/charm.py

```diff
@@ -110,6 +111,7 @@ def _on_collect_unit_status(self, event: CollectStatusEvent):
             event.add_status(WaitingStatus("Waiting for container to be ready"))
             logger.info("Waiting for container to be ready")
             return
+        self.unit.set_workload_version(self._get_workload_version())
```

Comment:
Why do we place this inside of collect unit status? It feels like we are adding a side effect to a method that had 1 purpose

Reply:
It is part of the status of the unit. Where else would it make sense to put it?

Reply:
The only other place is in configure, but we are not configuring the workload, we are reporting info about the charm. Would that be a better location?

Reply:
Yeah I agree that configure is probably not the right place either. Ok let's keep it here, maybe let's update the docstring to reflect that this hook also sets the workload version.